src/DelegateInterface.php

```diff
@@ -2,17 +2,17 @@
 
 namespace Interop\Http\Middleware;
 
-use Psr\Http\Message\RequestInterface;
 use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
 
 interface DelegateInterface
 {
     /**
-     * Dispatch the next available middleware and return the response.
+     * Process an incoming server request and return a response.
```

Comment:
> … because this method doesn't process … it merely delegates to something that does

*If* you are right, then the method name is wrong – it's **`::process()`** :stuck_out_tongue_winking_eye: 

From [Oxford Dictionaries](https://en.oxforddictionaries.com/definition/dispatch):
> Send off to a destination or for a purpose:
‘he dispatched messages back to base’

Thus, IMO, if you dispatch sth. you don't wait for a response. As you say – you're *delegating* the task, which is what I meant by *sounds asynchronous*. Of course it's not a must, but most Dispatchers I've seen in the wild mange processes, threads or some other kind of async tasks…